src/module/Common/src/Common/Filter/Shortify.php

```diff
@@ -13,21 +13,78 @@
+            'da' => true, 'dass' => true, 'dein' => true, 'deine' => true, 'dem' => true, 'den' => true, 'der' => true, 'des',
+            'dessen' => true, 'deshalb' => true, 'die' => true, 'dies' => true, 'dieser' => true, 'dieses' => true, 'doch' => true, 'dort',
+            'du' => true, 'durch' => true, 'ein' => true, 'eine' => true, 'einem' => true, 'einen' => true, 'einer' => true, 'eines' => true, 'er',
+            'es' => true, 'euer' => true, 'eure' => true, 'für' => true, 'hatte' => true, 'hatten' => true, 'hattest' => true, 'hattet',
+            'hier' => true, 'hinter' => true, 'ich' => true, 'ihr' => true, 'ihre' => true, 'im' => true, 'in' => true, 'ist',
+            'ja' => true, 'jede' => true, 'jedem' => true, 'jeden' => true, 'jeder' => true, 'jedes' => true, 'jener' => true, 'jenes',
+            'jetzt' => true, 'kann' => true, 'kannst' => true, 'können' => true, 'könnt' => true, 'machen' => true, 'mein' => true, 'meine',
+            'mit' => true, 'muß' => true, 'mußt' => true, 'musst' => true, 'müssen' => true, 'müßt' => true, 'nach' => true, 'nachdem',
+            'nein' => true, 'nicht' => true, 'nun' => true, 'oder' => true, 'seid' => true, 'sein' => true, 'seine' => true, 'sich',
+            'sie' => true, 'sind' => true, 'soll' => true, 'sollen' => true, 'sollst' => true, 'sollt' => true, 'sonst' => true, 'soweit',
+            'sowie' => true, 'und' => true, 'unser' => true, 'unsere' => true, 'unter' => true, 'vom' => true, 'von' => true, 'vor',
+            'wann' => true, 'warum' => true, 'was' => true, 'weiter' => true, 'weitere' => true, 'wenn' => true, 'wer' => true, 'werde',
+            'werden' => true, 'werdet' => true, 'weshalb' => true, 'wie' => true, 'wieder' => true, 'wieso' => true, 'wir',
+            'wird' => true, 'wirst' => true, 'wo' => true, 'woher' => true, 'wohin' => true, 'zu' => true, 'zum' => true, 'zur' => true, 'über'
+        ]
+    ];
+
+    /**
+     * @param string $word
+     * @return bool
+     */
+    static protected function isStopWord($word)
+    {
+        $lowerCased = strtolower($word);
+
+        foreach (self::$stopWords as $lang) {
+            if (array_key_exists($lowerCased, $lang)) return true;
+        }
+        return false;
+    }
+
     /**
      * @param string $text
      * @return bool|mixed
      */
     static protected function shortify($text)
     {
-        $text = preg_replace('@-(\w{1,3})-@isU', '-', $text);
-        $text = preg_replace('@-(\w{1,3})/@isU', '/', $text);
-        $text = preg_replace('@/(\w{1,3})-@isU', '/', $text);
+        $words = preg_split('/[\W]+/u', $text);
+
+        $filtered = [];
+
+        foreach ($words as $word) {
+            if (!self::isStopWord($word)) {
+                $filtered[] = $word;
+            }
+        }
 
-        if (empty($text)) {
+        if (empty($filtered)) {
             return false;
         }
 
-        return $text;
+        return implode(' ', $filtered);
```

Comment:
I think it would be smarter and less code to use a regexp here:

```
function($text) {
   // $pattern = getRegexpPattern();

   // The regexp is just a mock..you probably need to tweak it.
   // @([^\w]+|^)(aber|auch|du|der)([^\w]+|$)@isU

   return preg_replace($pattern, '\1\3', $text);
}
```

Using a regexp solves `implode(' ', $filtered)` which causes all non word-characters to be replaced by an empty space:
`PQ-Formel: Testnla` ->`PQ Formel Testnla`